cmd/zc_traverser_file.go

```diff
@@ -66,10 +68,28 @@ func (t *fileTraverser) getPropertiesIfSingleFile() (*azfile.FileGetPropertiesRe
 }
 
 func (t *fileTraverser) Traverse(preprocessor objectMorpher, processor objectProcessor, filters []ObjectFilter) (err error) {
+	invalidBlobOrWindowsName := func(path string) bool {
```

Comment:
TODO: should we only do this check when trailing dot is enabled? If explicitly disabled I think we can skip this error message

Reply:
Over in copyEnumeratorInit.go there is a function, `pathEncodeRules`. That feels like a decent place to be concerned about this instead of here? It'd be able to apply to more use-cases than just the files enumerator.

Reply:
Would that only apply to copy enumeration then? or also sync?

Reply:
Also adding this check in the traverser follows the behavior of S3 sources. PathEncodeRules seems more like a piece of code that encodes if a path is unsafe instead of skipping the path altogether

Reply:
NVM then. Trailing dot is something we only really have to deal with from S3 and from Files, anyway. It's fine to implement it as a specific thing for Files.